cmd/kubectl-gadget/utils/trace.go

```diff
@@ -597,7 +598,18 @@ func waitForCondition(traceID string, conditionFunction func(*gadgetv1alpha1.Tra
 	}
 
 	if err != nil {
-		return nil, err
+		if !errors.Is(err, wait.ErrWaitTimeout) {
+			return nil, err
+		}
+
+		// Print a message for traces that timed out
+		for _, trace := range traceList.Items {
+			_, satisfied := satisfiedTraces[trace.ObjectMeta.Name]
+			_, errored := erroredTraces[trace.ObjectMeta.Name]
+			if !satisfied && !errored {
+				fmt.Fprintf(os.Stderr, "Timeout waiting for condition on node %q\n", trace.Spec.Node)
+			}
+		}
```

Comment:
You should manage the case when all the traces timed out and return an error. Otherwise, the CLI will remain blocked, waiting to print events that will never arrive. I don't expect the CLI to be able to recover the communication with the node and resume the work when the node is up again. Or what did I miss?

Reply:
ops, you're right. I wanted to do it but got distracted and pushed the commit without this.